src/modules/popUp.js

```diff
@@ -8,64 +9,154 @@ class PopUp {
-    <div class="pop-up__container">
-        <img class="pop-up__container__img" src="${(await countryInfo).flagURl}" alt="${country.name}" />
-        <div class="pop-up__container__close-btn">
-            <i class="bi bi-x-circle"></i>
-        </div>
-        <div class="pop-up__container__content">
-            <h3 class="pop-up__container__content__title">${(await countryInfo).name}</h3>
-            <ul class="pop-up__container__content__details">
-                <li><span class="bold-text">Continent: </span><span class="span__br"><br></span>${(await countryInfo).continent}</li>
-                <li><span class="bold-text">Capital: </span><span class="span__br"><br></span>${(await countryInfo).capital}</li>
-                <li><span class="bold-text">Area: </span><span class="span__br"><br></span>${(await countryInfo).area} km2</li>
-             <li><span class="bold-text">Population: </span><span class="span__br"><br></span>${(await countryInfo).population}</li>
-            </ul>
-        </div>
-
-        <div class="pop-up__container__log">
-            <h3 class="pop-up__container__log__title">Comments</h3>
-            <ul class="pop-up__container__log__entries">
-                <li><span class="bold-text">24/06/2023 Alex:</span> I love that salad!</li>
-                <li><span class="bold-text">25/06/2023 Tania:</span> This dish is very delicious!</li>
-                <li><span class="bold-text">24/06/2023 Alex:</span> I love that salad!</li>
-                <li><span class="bold-text">24/06/2023 Alex:</span> I love that salad!</li>
-            </ul>
-        </div>
-
-        <div class="pop-up__container__form">
-            <h3 class="pop-up__container__form__title">Add a comment</h3>
-            <form action="">
-                <input type="text" placeholder="Name" />
-                <textarea name="" id="" cols="30" rows="10" placeholder="Your insights"></textarea>
-                <button class="pop-up__btn">Submit</button>
-            </form>
-        </div>
-    </div>
-  `;
-      const closeBtn = popUp.querySelector('.pop-up__container__close-btn');
-
-      closeBtn.addEventListener('click', () => {
-        popUp.classList.add('hidden');
-      });
+  createPopUp = async (country) => {
+    const countryInfo = this.countries.getCountryShortInfo(country);
+    const popUp = document.createElement('div');
+    popUp.classList.add('pop-up--active');
+
+    const popUpContent = document.createElement('div');
+    popUpContent.classList.add('pop-up__container');
+
+    const popUpImg = document.createElement('img');
+    popUpImg.classList.add('pop-up__container__img');
+    popUpImg.setAttribute('src', (await countryInfo).flagURl);
+    popUpImg.setAttribute('alt', country.name);
+    popUpContent.appendChild(popUpImg);
+
+    const popUpCloseBtn = document.createElement('div');
+    popUpCloseBtn.classList.add('pop-up__container__close-btn');
+    popUpCloseBtn.innerHTML = '<i class="bi bi-x-circle"></i>';
```

Comment:
- [x] Excellent on replacing the code from raw html to this structure. It is easier to handle some event listeners by this structure.